pkg/provisioningv2/rke2/planner/certificaterotation.go

```diff
@@ -86,11 +97,32 @@ func rotateCertificatesPlan(controlPlane *rkev1.RKEControlPlane, rotation *rkev1
 		}
 	}
 
+	rotateScriptPath := "/opt/rancher/provisioning/certificate_rotation/bin/rotate.sh"
+
+	args := []string{
+		"-xe",
+		rotateScriptPath,
+		rke2.GetRuntimeAgentUnit(controlPlane.Spec.KubernetesVersion),
+		strconv.FormatInt(rotation.Generation, 10),
+	}
+
+	if len(rotation.Services) > 0 {
+		for _, service := range rotation.Services {
+			args = append(args, "-s", service)
+		}
+	}
+
 	return plan.NodePlan{
+		Files: []plan.File{
+			{
+				Content: base64.StdEncoding.EncodeToString([]byte(idempotentRotateScript)),
```

Comment:
Question - this script is meant to be run programmatically right? Or manually? If it's the former(which I think it is), where do you pass the current and target gen in as args?

Reply:
The file is copied, but it is executed through the planner as part of the `OneTimeInstruction.Args`, so it's easy to miss: https://github.com/rancher/rancher/blob/b1b5da4362e5a4057c085cdae0ec644d929d3cfd/pkg/provisioningv2/rke2/planner/certificaterotation.go#L104

Reply:
I see, thanks!